### ISSUE TITLE
donut使用的问题

### ISSUE THREAD

**Issue by yang182** [haowen-xu/donut#7]
您好
我在用基于vae的donut算法做异常检测，但是这个模型只能对于每一个曲线训练一个模型，而我有很多曲线需要训练，这样我就需要训练许多模型。我能写一个循环对于每一条曲线训练一个模型吗
我如何将提前训练好的模型加载直接用来预测，而不用再训练

**Comment by lizeyan** [comment]
我个人的经验是最好不要用循环训练很多模型，而是每训练一个模型都单独起一个进程。
怎么保存和加载模型readme里讲了。

**Comment by haowen-xu** [comment]
虽然你可以 with tf.Session()，每个模型用一个新的 Session，在同一个进程里面处理多个数据集。但是我建议你把训练和测试写成单独的脚本，用命令行参数控制训练数据和模型持久化，然后用另一个批处理脚本做循环。

**Comment by polarbear1992** [comment]
> 虽然你可以 with tf.Session()，每个模型用一个新的 Session，在同一个进程里面处理多个数据集。但是我建议你把训练和测试写成单独的脚本，用命令行参数控制训练数据和模型持久化，然后用另一个批处理脚本做循环。

你好 Haowen学长，我最近也在用donut算法做异常检测，现在的情况是我已经训练出来了多个模型，但是对于实时数据我需要跟据数据的类别分发至对应的模型进行检测，所以我需要让多个模型并行化，我在每个session里加载一个模型，然后在启用这个session对数据进行异常检测，但是会在最后的预测中获取predictor.get_score时报错：
![image](https://user-images.githubusercontent.com/16411944/71898908-78839980-3195-11ea-9a90-1e2b239f1b69.png)

我的代码如下，能否帮忙看一下是哪里出了问题~~~
    
    g1 = tf.Graph()
    g2 = tf.Graph()
    sess1 = tf.Session(graph=g1)
    sess2 = tf.Session(graph=g2)
    
    with sess1.as_default():
        with g1.as_default():
            with tf.variable_scope('model') as model_vs:
                    model = Donut(
                            h_for_p_x=Sequential([
                                K.layers.Dense(100, kernel_regularizer=K.regularizers.l2(0.001),
                                               activation=tf.nn.relu),
                                K.layers.Dense(100, kernel_regularizer=K.regularizers.l2(0.001),
                                               activation=tf.nn.relu),
                            ]),
                            h_for_q_z=Sequential([
                                K.layers.Dense(100, kernel_regularizer=K.regularizers.l2(0.001),
                                               activation=tf.nn.relu),
                                K.layers.Dense(100, kernel_regularizer=K.regularizers.l2(0.001),
                                               activation=tf.nn.relu),
                            ]),
                            x_dims=120,
                            z_dims=5,
                        )
            trainer = DonutTrainer(model=model, model_vs=model_vs)
            predictor = DonutPredictor(model)
            save_dir = parent_path_model + path_model
            saver=VariableSaver(get_variables_as_dict(model_vs), save_dir,\
                            filename="cluster_" + "1" +"_data" + "_" + 'variables.dat',\
                            latest_file='latest_' + "cluster_" + "1" +"_data")
            saver.restore()

    with sess2.as_default():
        with g2.as_default():
            with tf.variable_scope('model') as model_vs:
                    model = Donut(
                            h_for_p_x=Sequential([
                                K.layers.Dense(100, kernel_regularizer=K.regularizers.l2(0.001),
                                               activation=tf.nn.relu),
                                K.layers.Dense(100, kernel_regularizer=K.regularizers.l2(0.001),
                                               activation=tf.nn.relu),
                            ]),
                            h_for_q_z=Sequential([
                                K.layers.Dense(100, kernel_regularizer=K.regularizers.l2(0.001),
                                               activation=tf.nn.relu),
                                K.layers.Dense(100, kernel_regularizer=K.regularizers.l2(0.001),
                                               activation=tf.nn.relu),
                            ]),
                            x_dims=120,
                            z_dims=5,
                        )
            trainer = DonutTrainer(model=model, model_vs=model_vs)
            predictor = DonutPredictor(model)
            save_dir = parent_path_model + path_model
            saver=VariableSaver(get_variables_as_dict(model_vs), save_dir,\
                            filename="cluster_" + str(2) +"_data" + "_" + 'variables.dat',\
                            latest_file='latest_' + "cluster_" + str(2) +"_data")
            saver.restore()


    with sess1.as_default():    
        with sess1.graph.as_default():        
            df = pd.read_csv("E:\\智能告警\\测试集-expand5min\\expand5min_obj_0data.csv")     
            tmp_timestamps = [time.mktime(time.strptime(t,"%Y-%m-%d %H:%M:%S")) for \                         
                                        t in list(df['time'])]         
            df['timestamp'] = tmp_timestamps        
            df.sort_values(by='timestamp',axis=0,ascending=True)        
            values = [float(x) for x in list(df['data'])]        
            timestamps = list(df['timestamp'])        
            label = np.zeros_like(values, dtype=np.int32)
            #缺失值补充
            timestamps,missing,(interp_data,label) = linear_interpolation(\
                               timestamps,(values,label),mode = False)
    
            #标准化至均值0、标准差1
            std_data,mean,std = standardize_obj(interp_data)
            scores = predictor.get_score(std_data,missing)

**Comment by fffii** [comment]
您好，请问如果一个数据集有很多条序列，且长度不同，那么改如何处理呢

**Comment by fffii** [comment]
您好，请问如果一个数据集有很多条序列，且长度不同，那么该如何处理呢
